diff --git a/include/sleepy_discord/client.h b/include/sleepy_discord/client.h
index fc190c0..379e29f 100644
--- a/include/sleepy_discord/client.h
+++ b/include/sleepy_discord/client.h
@@ -442,6 +442,7 @@ namespace SleepyDiscord {
 		const bool isReady() { return ready; }
 		const bool isQuiting() { return quiting; }
 		const bool isBot() { return bot; }
+		const bool isRateLimited() { return messagesRemaining <= 0 || request(Get, "gateway").statusCode == TOO_MANY_REQUESTS; }
 		const Snowflake<User> getID() { return userID; }
 		void setShardID(int _shardID, int _shardCount); //Note: must be called before run or reconnect
 		const int getShardID() { return shardID; }
