diff --git a/src/msvcbuild.bat b/src/msvcbuild.bat
index f1c63de9..5e68f0f8 100644
--- a/src/msvcbuild.bat
+++ b/src/msvcbuild.bat
@@ -14,8 +14,10 @@
 
 @setlocal
 @rem Add more debug flags here, e.g. DEBUGCFLAGS=/DLUA_USE_APICHECK
-@set DEBUGCFLAGS=
-@set LJCOMPILE=cl /nologo /c /O2 /W3 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_STDIO_INLINE=__declspec(dllexport)__inline
+@set DEBUGCFLAGS=/Od
+@rem Add more release flags here, e.g. /O2
+@set RELEASECFLAGS=/O2
+@set LJCOMPILE=cl /nologo /c /W3 /D_CRT_SECURE_NO_DEPRECATE /Z7
 @set LJLINK=link /nologo
 @set LJMT=mt /nologo
 @set LJLIB=lib /nologo /nodefaultlib
@@ -89,11 +91,14 @@ buildvm -m vmdef -o jit\vmdef.lua %ALL_LIB%
 buildvm -m folddef -o lj_folddef.h lj_opt_fold.c
 @if errorlevel 1 goto :BAD
 
-@if "%1" neq "debug" goto :NODEBUG
+@if "%1" neq "debug" goto :RELEASE
 @shift
 @set BUILDTYPE=debug
-@set LJCOMPILE=%LJCOMPILE% /Zi %DEBUGCFLAGS%
+@set LJCOMPILE=%LJCOMPILE% %DEBUGCFLAGS%
 @set LJLINK=%LJLINK% /opt:ref /opt:icf /incremental:no
+goto :NODEBUG
+:RELEASE
+@set LJCOMPILE=%LJCOMPILE% %RELEASECFLAGS%
 :NODEBUG
 @if "%1"=="amalg" goto :AMALGDLL
 @if "%1"=="static" goto :STATIC
